concent_api/core/message_handlers.py

```diff
@@ -1354,6 +1366,9 @@ def handle_send_subtask_results_verify(
             reason=message.concents.ServiceRefused.REASON.InvalidRequest,
         )
 
+    if Subtask.objects.filter(subtask_id=compute_task_def['subtask_id']).exists():  # pylint: disable=no-member
+        Subtask.objects.select_for_update().get(subtask_id=compute_task_def['subtask_id'])  # pylint: disable=no-member
+
     if Subtask.objects.filter(  # pylint: disable=no-member
```

Comment:
It would be better if this function received `subtask` in a parameter instead of fetching it on its own.

As it is now, the lock is hidden in the validator and not well visible. I think the caller (view/handler) should be responsible for locking it before using the validator.

Reply:
I really don't understand.. :/ I think i did it like this at the beginning and you asked to change it to create as small atomic blocks as possible. The difference was exactly about this validation, I think you asked they should be before atomic and lock.. Or am I wrong?